core/models/access.py

```diff
@@ -38,6 +38,33 @@ class Meta:
             )
         ]
 
+    @classmethod
+    def expire_accesses(cls, upper_date: date) -> None:
+        """
+        Set the status of Access objects with an expiration date lower than given `upper_date`
+        to terminated
+
+        @param upper_date: Date threshold.
+        """
+        accesses_to_expire = cls.objects.filter(
+            grant_expires_on__lt=upper_date
+        ).exclude(
+            status=StatusChoices.terminated,
+        ).all()
+        for access in accesses_to_expire:
```

Comment:
would be good to add some logging messages

Reply:
Done.
I added one debug log in the loop to mention accesses that are expired and to show the comparison between the expiration date and the given parameter.

I added another one at the end to say that the change was done successfully.